### PR TITLE
Add MachineSet infra and server group content

### PR DESCRIPTION
**Pull request by maxwelldb**
- Add OSP-specific MachineSet example to infrastructure docs assembly
- Add server group property to sample YAML
- Add server group property guidance to sample callouts

Context:
- https://bugzilla.redhat.com/show_bug.cgi?id=1858989
- https://issues.redhat.com/browse/OSDOCS-1364
- https://github.com/openshift/installer/pull/4016/

@pierreprinetti Does this look copacetic to you?

### CODE REVIEW

**Comment by maxwelldb** [comment]
https://aa-workers-osdocs1364--ocpdocs.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets.html#machineset-yaml-osp_creating-infrastructure-machinesets

File: modules/machineset-yaml-osp.adoc [hunk 1]
```diff
@@ -78,3 +79,4 @@ $ oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster
 ----
 <2> Specify the node label to add.
 <3> Specify the infrastructure ID and node label.
+<4> Enter the value that is returned when link:https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/16.0/html/command_line_interface_reference/server#server_group_create[creating a server group]. The server group must have `anti-affinity` or `soft-anti-affinity` policies.
```
Comment (pierreprinetti) [inline modules/machineset-yaml-osp.adoc:82]: Actually, while I can't imagine a reason for setting `affinity` or `soft-affinity`, any policy (or none) is fine as far as OpenShift is concerned.

Comment (pierreprinetti) [reply modules/machineset-yaml-osp.adoc:82]: I'd go like:
"Can be used to set an anti-affinity policy for the MachineSet".

**Comment by maxwelldb** [comment]
@pierreprinetti How about https://github.com/openshift/openshift-docs/pull/25837/commits/83aef7895bc2af20733be288f58e9ae5f8a659ff, then, if any policy works? 

Would it make sense to say that we recommend anti-affinity or soft-anti-affinity policies for most deployments, or is that too prescriptive?

**Comment by pierreprinetti** [comment]
> or is that too prescriptive?

No Sir, that would be copacetic.

**Comment by maxwelldb** [comment]
@pierreprinetti TY. Added recommendation in 8838967.

**Review by pierreprinetti — APPROVED** [review]
LGTM 👍

**Comment by maxwelldb** [comment]
Thanks! @pierreprinetti

**Review by kalexand-rh — COMMENTED** [review]
LGTM

**Comment by maxwelldb** [comment]
Thanks!

**Pull request merged**